### PR TITLE
Add super admin check

### PR DESCRIPTION
**Pull request by trevrosen**
## What
Implement a check for super admin

### CODE REVIEW

**Comment by trevrosen** [comment]
@jspies can you explain why you did this here? I copied on my new function in this PR and @B-3PO asked about it. I'm not getting why throw and then immediately catch an exception in the same scope. https://github.com/bypasslane/bypass-passport-strategy/blob/master/lib/bypass_auth.js#L54-L64

cc @sdrioux

File: lib/bypass_auth.js [hunk 1]
```diff
@@ -63,6 +63,18 @@ BypassAuth.restrictToAdmin = function(req, res, next) {
   }
 };
 
+BypassAuth.restrictToSuperAdmin = function(req, res, next) {
+  try {
+    if (!req.user || req.user.account.super_admin !== true) {
+     throw new Error("Must be a super admin");
+    }
+
+    return next();
```
Comment (trevrosen) [inline lib/bypass_auth.js]: @B-3PO thanks for the feedback. `return next()` is used throughout this file -- are you saying it's a mistake?

File: lib/bypass_auth.js [hunk 2]
```diff
@@ -63,6 +63,14 @@ BypassAuth.restrictToAdmin = function(req, res, next) {
   }
 };
 
+BypassAuth.restrictToSuperAdmin = function(req, res, next) {
+    if (!req.user || req.user.account.super_admin !== true) {
```
Comment (dvictory) [inline lib/bypass_auth.js]: what is user.account = undefined ?

Comment (dvictory) [inline lib/bypass_auth.js]: Do we need to check that, or can we assume all users have an account?

**Comment by texasjusticar** [comment]
🚢

**Comment by txssseal** [comment]
👍

**Pull request merged**